### ISSUE TITLE
DTMF command?

### ISSUE THREAD

**Issue by benlangfeld** [rayo/xmpp#32]
I wonder if we need a DTMF command, or if this should simply be wrapped up in the output component. Thoughts?

**Comment by benlangfeld** [comment]
@mpermar: Yeah, the only advantage I can think of about having it is that it is self-explanatory. Otherwise you would have to enter into explaining the output syntax needed to send dtmf tones. 

I think either is equal effort to specify, and using output keeps it more generic, so that would be my preference. Ticket here: https://github.com/rayo/xmpp/issues/32

**Comment by benlangfeld** [comment]
Two considerations:
- DTMF tones should not be contained within an SSML document because:
  1. There is no dtmf tag in SSML.
  2. Not all media servers are capable of parsing something like `<audio src="dtmf-1"/>`, or implement some variant.
- DTMF is a special case of a general tone, and is just output. As such, it should stay in the output component.

Combining these, I can only conclude that DTMF should be an output component with a document type specifying it as a DTMF sequence, with a specified format. Other tones may be rendered similarly.

**Comment by benlangfeld** [comment]
So, we need output to have a <document> child similar to <grammar> in input, which has a content type attribute. We also need to define a document format for DTMF output. @loopingrage Are any such formats already defined, or do you have any thoughts on how we might do this?

**Comment by loopingrage** [comment]
Voxeo's CCXML browser uses application/x-senddtmf http://docs.voxeo.com/ccxml/1.0-final/frame.jsp?page=dialogtype_senddtmf10.htm but I've never been a fan of the format. Need to think about this one.

**Comment by crienzo** [comment]
A standard DTMF document format would help with making the functional test suite implementation-independent.

**Comment by benlangfeld** [comment]
Does anyone know where x-senddtmf is documented, or if there are any better documented formats? Do we need to define one? If we do define one, either an extension or a parallel spec to SSML would be my preference.

**Comment by benlangfeld** [comment]
@crienzo suggested using [FreeSWITCH's tone format](http://wiki.freeswitch.org/wiki/TGML) as a starting point. @burnburn mentioned that [WebRTC is defining DTMF formats](http://dev.w3.org/2011/webrtc/editor/webrtc.html#peer-to-peer-dtmf) and that the W3C Voice Browser Working Group would be a good place for this spec.

**Comment by chewi** [comment]
Perhaps you missed [XEP-0181](http://xmpp.org/extensions/xep-0181.html)? It's deferred so I almost missed it myself but being an XEP, I guess it's a little closer to home than the other suggestions.

**Comment by benlangfeld** [comment]
@burnburn mentioned at AdhearsionConf in Dec 2013 that WebRTC would adopt a stand-alone DTMF document format if one existed, and I _think_ I remember he suggested that the IETF would be the appropriate place for that. Something along the lines of XEP-0181 might be acceptable, but it could not be an XEP or jingle-specific to meet these requirements.

**Comment by burnburn** [comment]
I believe the proper statement now is that the WebRTC group _would have_ adopted a stand-alone DTMF document format if one had existed.  At this point (July 2014) I think it unlikely any change will occur for the format defined in the WebRTC spec, at least for version 1.0 of WebRTC.

However, it may still be worth pursuing defining such a format.  IETF is a reasonable candidate to do such work, but you will have to be able to explain why it's not more appropriate to do it in a traditional telco SDO such as ITU or even 3GPP/GSMA.  I don't have a good answer off the top of my head.
